### PR TITLE
[SQL] Make 'emitHandles' an internal compiler options

### LINKED ISSUE #1151: RUST ERROR creating SQL Program
Following DDL results in 
`CREATE TABLE event_t (
id BIGINT NOT NULL PRIMARY KEY,
local_event_dt DATE
);`

image

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

Fixes #1151 

We generate code in different ways for the pipeline and for testing locally. This centralizes this information into a new "internal" compiler flag, `emitHandles`. The flag is internal because it is not exposed as a compiler command-line option. It can be only modified programmatically.

I have moved several tests which need this flag into a common file, the testing infrastructure makes sure that you can only run together tests that have the same compiler flags.

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/ComplexQueriesTest.java [hunk 1]
```diff
@@ -121,311 +114,6 @@ public void smallTaxiTest() {
         this.addRustTestCase("ComplexQueriesTest.smallTaxiTest", compiler, getCircuit(compiler));
     }
 
-    // Also compiles the code using the Rust handle API
-    @Test
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/ComplexQueriesTest.java:125]: this test was almost identical with the next one, removed the simpler copy and moved the other one to a separate file

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/DBSPCompilerTests.java [hunk 2]
```diff
@@ -53,27 +42,6 @@ public class DBSPCompilerTests {
             ", COL4 VARCHAR NOT NULL" +
             ")";
 
-    @Test
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/DBSPCompilerTests.java:56]: moved to a separate file

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/HandlesTests.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+public class HandlesTests extends BaseSQLTests {
+    @Override
+    public CompilerOptions testOptions(boolean incremental, boolean optimize) {
+        CompilerOptions result = super.testOptions(incremental, optimize);
+        result.ioOptions.emitHandles = true;
+        return result;
+    }
+
+    @Test
+    public void testSanitizeNames() throws IOException, InterruptedException {
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/HandlesTests.java:22]: most of this code was moved from other files

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/OtherTests.java [hunk 4]
```diff
@@ -634,33 +633,6 @@ public void testFreshName() {
         Assert.assertEquals(t1, "T_1");
     }
 
-    @Test
-    public void testSanitizeNames() throws IOException, InterruptedException {
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/OtherTests.java:638]: moved to another file

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**